### PR TITLE
[DO NOT MERGE] create an action for deploying an awscli helm chart

### PR DESCRIPTION
**Pull request by msnwatson**
Associated tickets or Slack threads:
- #1845 

## How does this fix it?[^1]
* I want to temporarily create a new pod with the AWS CLI installed so I can interact with the RDS dbinstances more easily while developing an application to do the DB startup steps programmatically in the future. I tried creating a deployment of the container referenced in the helm chart using basic kubectl commands as well as looked for a pre-configured helm chart i could deploy but found that nothing worked. 

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by yoomlam — APPROVED** [review]

**Comment by yoomlam** [comment]
@msnwatson I don't think you even need to merge this to `develop`. Just run all the GH Action workflows (SecRel, update-deployment) with your branch selected. In your branch, you'll need to add an `awscli` entry to the options: https://github.com/department-of-veterans-affairs/abd-vro/blob/5bb3a5bfd419a91b797e84b6bc440940629db0f9/.github/workflows/update-deployment.yml#L23C1-L23C1

File: .github/workflows/update-deployment-awscli.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        type: boolean
+        default: false
+
+jobs:
+  update-deployment-platform:
+    uses: ./.github/workflows/update-deployment.yml
+    secrets: inherit
+    with:
+      target_env: ${{ inputs.target_env }}
+      helm_chart: awscli
```
Comment (yoomlam) [inline .github/workflows/update-deployment-awscli.yml:51]: This won't work unless you also add an `awscli` entry to the options: 
https://github.com/department-of-veterans-affairs/abd-vro/blob/5bb3a5bfd419a91b797e84b6bc440940629db0f9/.github/workflows/update-deployment.yml#L23C1-L23C1

**Comment by yoomlam** [comment]
> I tried creating a deployment of the container referenced in the helm chart using basic kubectl commands as well as looked for a pre-configured helm chart i could deploy but found that nothing worked.

To deploy, I use helm on my local machine; for example (from my shell history):
```
> helm upgrade vro-svc-bie-kafka helm/svc-bie-kafka -n va-abd-rrd-dev --install --reset-values --set-string global.imageTag=latest --set-string global.commitSha=10f509cc947dfb044d5b63dcdfc03168b4cac1b3 -f helm/_shared/values.yaml -f helm/_shared/values-for-dev.yaml -f helm/svc-bie-kafka/values.yaml --debug | less
```
like what is done [here](https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/5967823027/job/16190324712#step:6:83)

**Comment by msnwatson** [comment]
> > I tried creating a deployment of the container referenced in the helm chart using basic kubectl commands as well as looked for a pre-configured helm chart i could deploy but found that nothing worked.
> 
> To deploy, I use helm on my local machine; for example (from my shell history):
> 
> ```
> > helm upgrade vro-svc-bie-kafka helm/svc-bie-kafka -n va-abd-rrd-dev --install --reset-values --set-string global.imageTag=latest --set-string global.commitSha=10f509cc947dfb044d5b63dcdfc03168b4cac1b3 -f helm/_shared/values.yaml -f helm/_shared/values-for-dev.yaml -f helm/svc-bie-kafka/values.yaml --debug | less
> ```
> 
> like what is done [here](https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/5967823027/job/16190324712#step:6:83)

and i'm guessing this will use the current kubeconfig for the shell to choose which k8s cluster to run the operation on?

**Comment by yoomlam** [comment]
> and i'm guessing this will use the current kubeconfig for the shell to choose which k8s cluster to run the operation on?

Yes, specify the kubeconfig like this:
```
> helm --kubeconfig=$HOME/.kube/config-prod
```

**Comment by yoomlam** [comment]
Yup! No need to merge: go to https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/workflows/update-deployment.yml and select your branch:
image

File: .github/workflows/update-deployment-awscli.yml
Comment (yoomlam) [inline .github/workflows/update-deployment-awscli.yml:1]: Don't need this file.

File: helm/awscli/values.yaml
Comment (yoomlam) [inline helm/awscli/values.yaml:1]: You don't need this file if your `deployment.yaml` doesn't reference these values.

**Pull request closed without merging**